client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-add-license-notification/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { useTranslate } from 'i18n-calypso';
+import { ReactElement, useEffect, useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import Banner from 'calypso/components/banner';
+import { setPurchasedProduct } from 'calypso/state/jetpack-agency-dashboard/actions';
+import type { PurchasedProduct } from '../types';
+
+import './style.scss';
+
+export default function SiteAddLicenseNotification( {
+	purchasedLicense,
+}: {
+	purchasedLicense: PurchasedProduct;
+} ): ReactElement | null {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const selectedSite = purchasedLicense.selectedSite;
+	const selectedProduct = purchasedLicense.selectedProduct;
+
+	const dismissBanner = useCallback( () => {
+		dispatch( setPurchasedProduct() );
+	}, [ dispatch ] );
+
+	useEffect( () => {
+		return () => {
+			dismissBanner();
+		};
+	}, [ dismissBanner ] );
```

Comment:
What's the purpose of this `useEffect`? Why do we need to dismiss the banner when the component is deleted/garbage collected?

Reply:
This is to clear the notification when the user navigated to other page and comes back to dashboard